<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1085</width>
    <height>872</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="RawDataLoad">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>10</y>
      <width>449</width>
      <height>139</height>
     </rect>
    </property>
    <property name="title">
     <string> Data Load</string>
    </property>
    <widget class="QLabel" name="path_print">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::Box</enum>
     </property>
     <property name="text">
      <string>Select file path</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="path_btn">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>30</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Path</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="path_edit">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>70</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Input data file path edit</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="load">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>70</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Load</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="header_row">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>110</y>
       <width>50</width>
       <height>20</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="frame">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="headerRow_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>91</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Header Row</string>
     </property>
    </widget>
    <widget class="QLabel" name="indexCol_label">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>110</y>
       <width>101</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Index column</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="indexCol_edit">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>110</y>
       <width>161</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="factorAnalysis">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>160</y>
      <width>451</width>
      <height>321</height>
     </rect>
    </property>
    <property name="title">
     <string> Exploratory Factor Analysis</string>
    </property>
    <widget class="QPushButton" name="fitting_btn">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>280</y>
       <width>411</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Fitting</string>
     </property>
    </widget>
    <widget class="QLabel" name="faNum_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Factor Number</string>
     </property>
    </widget>
    <widget class="QLabel" name="rotation_label">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>30</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Rotation Method</string>
     </property>
    </widget>
    <widget class="QComboBox" name="rotation_comboBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>50</y>
       <width>271</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>None</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>varimax (orthogonal rotation)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>promax (oblique rotation)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>oblimin (oblique rotation)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>oblimax (orthogonal rotation)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>quartimin (oblique rotation)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>quartimax (orthogonal rotation)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>equamax (orthogonal rotation)</string>
      </property>
     </item>
    </widget>
    <widget class="QSpinBox" name="fa_num_spin">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="value">
      <number>3</number>
     </property>
    </widget>
    <widget class="QComboBox" name="fitting_comboBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>110</y>
       <width>271</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>minres (Minimal Residual Method)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>ml (Maximum Likehood Estimate)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>principal (Principal Component Analysis)</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="fitting_label">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>90</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Fitting Method</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="use_smc_ckbox">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>220</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>use_smc</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QCheckBox" name="is_corr_mat_ckbox">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>220</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>is_corr_matrix</string>
     </property>
    </widget>
    <widget class="QComboBox" name="impute_comboBox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>110</y>
       <width>121</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>median</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>mean</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>drop</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="impute_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>121</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>miss data impute</string>
     </property>
    </widget>
    <widget class="QComboBox" name="fitting_comboBox_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>411</width>
       <height>31</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string>randomized (faster randomized SVD from scikit-learn)</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>lapack (standard SVD from scipy)</string>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="fitting_label_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>411</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>If the fitting method is PCA, select the decomposition method</string>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="Upper_bound">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>220</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="decimals">
      <number>3</number>
     </property>
     <property name="singleStep">
      <double>0.005000000000000</double>
     </property>
     <property name="value">
      <double>1.000000000000000</double>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="Lower_bound">
     <property name="geometry">
      <rect>
       <x>80</x>
       <y>250</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="decimals">
      <number>3</number>
     </property>
     <property name="minimum">
      <double>0.005000000000000</double>
     </property>
     <property name="maximum">
      <double>1.000000000000000</double>
     </property>
     <property name="singleStep">
      <double>0.005000000000000</double>
     </property>
     <property name="value">
      <double>0.005000000000000</double>
     </property>
    </widget>
    <widget class="QLabel" name="bound_label">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>200</y>
       <width>67</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Bound</string>
     </property>
    </widget>
    <widget class="QLabel" name="upper_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>220</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Upper </string>
     </property>
    </widget>
    <widget class="QLabel" name="lower_label">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>250</y>
       <width>51</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>11</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Lower</string>
     </property>
    </widget>
    <widget class="QLabel" name="otherOP_label">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>200</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>other options</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>620</x>
      <y>490</y>
      <width>451</width>
      <height>331</height>
     </rect>
    </property>
    <property name="title">
     <string>  Result Output</string>
    </property>
    <widget class="QCheckBox" name="fa_loading_ckbox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>91</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Loadings</string>
     </property>
    </widget>
    <widget class="QLabel" name="result_output_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>161</width>
       <height>17</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Fitting Result Output</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="fa_comm_ckbox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>50</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Communalities</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="fa_eigenv_ckbox">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>50</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Eigenvalues</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="fa_var_ckbox">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>90</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>factor variance</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="fa_unique_ckbox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>90</y>
       <width>141</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>uniquenesses</string>
     </property>
    </widget>
    <widget class="QLabel" name="suffi_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>130</y>
       <width>171</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Sufficiency :  obs num</string>
     </property>
    </widget>
    <widget class="QPushButton" name="suffi_do_btn">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>130</y>
       <width>161</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Do</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="suffi_obs_num_spin">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>130</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
    </widget>
    <widget class="QPushButton" name="transform_btn">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>280</y>
       <width>151</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Transform</string>
     </property>
    </widget>
    <widget class="QLabel" name="suffi_label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>170</y>
       <width>271</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Get factor scores for a new data set</string>
     </property>
    </widget>
    <widget class="QPushButton" name="path_btn_3">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>200</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Path</string>
     </property>
    </widget>
    <widget class="QLabel" name="path_print_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>200</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::Shape::Box</enum>
     </property>
     <property name="text">
      <string>Select file path</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLineEdit" name="path_edit_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>237</y>
       <width>341</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Input data file path</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
    <widget class="QPushButton" name="load_2">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>237</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Load</string>
     </property>
    </widget>
    <widget class="QPushButton" name="show_result">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>90</y>
       <width>131</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Show</string>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>601</width>
      <height>811</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QGroupBox" name="DataFrameView">
       <property name="title">
        <string>  Data Frame View</string>
       </property>
       <widget class="QTableWidget" name="DataViewTable">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>581</width>
          <height>241</height>
         </rect>
        </property>
       </widget>
       <widget class="QGroupBox" name="result">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>280</y>
          <width>599</width>
          <height>391</height>
         </rect>
        </property>
        <property name="title">
         <string>  Result View</string>
        </property>
        <widget class="QTextBrowser" name="result_out">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>581</width>
           <height>311</height>
          </rect>
         </property>
        </widget>
        <widget class="QPushButton" name="clear_faout_btn">
         <property name="geometry">
          <rect>
           <x>450</x>
           <y>350</y>
           <width>131</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </widget>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>680</y>
          <width>581</width>
          <height>121</height>
         </rect>
        </property>
        <property name="title">
         <string>Export</string>
        </property>
        <widget class="QPushButton" name="result_to_excel_btn">
         <property name="geometry">
          <rect>
           <x>440</x>
           <y>30</y>
           <width>131</width>
           <height>31</height>
          </rect>
         </property>
         <property name="text">
          <string>result to excel</string>
         </property>
        </widget>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1085</width>
     <height>27</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
